ray-operator/controllers/ray/common/service.go

```diff
@@ -184,29 +151,67 @@ func BuildServeServiceForRayService(rayService rayv1alpha1.RayService, rayCluste
 		RayClusterServingServiceLabelKey: EnableRayClusterServingServiceTrue,
 	}
 
-	service := &corev1.Service{
+	default_name := utils.GenerateServeServiceName(rayService.Name)
+	default_namespace := rayService.Namespace
+	default_type := rayService.Spec.RayClusterSpec.HeadGroupSpec.ServiceType
+
+	// `ports_int` is a map of port names to port numbers, while `ports` is a list of ServicePort objects
+	ports_int := getServicePorts(rayCluster)
+	ports := []corev1.ServicePort{}
+	for name, port := range ports_int {
+		if name == DefaultServingPortName {
+			svcPort := corev1.ServicePort{Name: name, Port: port}
+			ports = append(ports, svcPort)
+			break
+		}
+	}
+
+	if rayService.Spec.ServeService != nil {
+		// Use the provided "custom" ServeService.
+		// Deep copy the ServeService to avoid modifying the original object
+		serveService := rayService.Spec.ServeService.DeepCopy()
+
+		// For the selector, ignore any custom ServeService selectors or labels.
+		serveService.Spec.Selector = selectorLabels
+
+		// Add DeafultServePort if it is already not added and ignore any custom ports
+		// Keeping this consistentent with adding only serve port in serve service
```

Comment:
```suggestion
		// Add DefaultServePort if it is already not added and ignore any custom ports
		// Keeping this consistent with adding only serve port in serve service
```

Reply:
Does it make sense to print or log a warning when we see a custom port and ignore it?

Reply:
I can add the warning.